docs/MatrixOne/Reference/SQL-Reference/Data-Manipulation-Language/0.8-insert-on-duplicate.md

```diff
@@ -0,0 +1,66 @@
+# **INSERT ... ON DUPLICATE KEY UPDATE**
+
+## **Description**
+
+`INSERT ... ON DUPLICATE KEY UPDATE` is used to insert data into the database table; if the data already exists, update the data; otherwise, insert new data.
+
+The `INSERT INTO` statement is a standard statement used to insert data into a database table; the `ON DUPLICATE KEY UPDATE` statement performs an update operation when there are duplicate records in the table. If a record with the same unique index or primary key exists in the table, use the `UPDATE` clause to update the corresponding column value; otherwise, use the `INSERT` clause to insert a new record.
+
+It should be noted that the premise of using this syntax is that a primary key constraint needs to be established in the table to determine whether there are duplicate records. At the same time, both the update operation and the insert operation need to set the corresponding column value. Otherwise, a syntax error will result.
```

Comment:
primary key or unique key

Reply:
@ouyuanning  confirmed with me here, and @fengttt  suggested supporting the primary key at the meeting. Because unique keys can be null, some troublesome problems could be solved.